web/src/model/topology.ts

```diff
@@ -37,94 +39,15 @@ export enum LayoutName {
-  zones = 'zones',
-  zonesHosts = 'zones+hosts',
-  zonesNamespaces = 'zones+namespaces',
-  zonesOwners = 'zones+owners',
-  hosts = 'hosts',
-  hostsNamespaces = 'hosts+namespaces',
-  hostsOwners = 'hosts+owners',
-  namespaces = 'namespaces',
-  namespacesOwners = 'namespaces+owners',
-  owners = 'owners'
-}
-
-export const getGroupsForScope = (scope: MetricScopeOptions) => {
-  switch (scope) {
-    case MetricScopeOptions.CLUSTER:
-      return [TopologyGroupTypes.none];
-    case MetricScopeOptions.ZONE:
-      return [TopologyGroupTypes.none, TopologyGroupTypes.clusters];
-    case MetricScopeOptions.HOST:
-      return [TopologyGroupTypes.none, TopologyGroupTypes.clusters, TopologyGroupTypes.zones];
-    case MetricScopeOptions.NAMESPACE:
-      return [
-        TopologyGroupTypes.none,
-        TopologyGroupTypes.clusters,
-        TopologyGroupTypes.clustersHosts,
-        TopologyGroupTypes.zones,
-        TopologyGroupTypes.zonesHosts,
-        TopologyGroupTypes.hosts
-      ];
-    case MetricScopeOptions.OWNER:
-      return [
-        TopologyGroupTypes.none,
-        TopologyGroupTypes.clusters,
-        TopologyGroupTypes.clustersZones,
-        TopologyGroupTypes.zones,
-        TopologyGroupTypes.zonesHosts,
-        TopologyGroupTypes.zonesNamespaces,
-        TopologyGroupTypes.zonesOwners,
-        TopologyGroupTypes.hosts,
-        TopologyGroupTypes.hostsNamespaces,
-        TopologyGroupTypes.namespaces
-      ];
-    case MetricScopeOptions.RESOURCE:
-    default:
-      return Object.values(TopologyGroupTypes);
-  }
-};
-
-export const isGroupEnabled = (group: TopologyGroupTypes, enabledScopes: FlowScope[]): boolean => {
-  return (
-    (enabledScopes.includes('cluster') || !group.includes(TopologyGroupTypes.clusters)) &&
-    (enabledScopes.includes('zone') || !group.includes(TopologyGroupTypes.zones)) &&
-    (enabledScopes.includes('host') || !group.includes(TopologyGroupTypes.hosts)) &&
-    (enabledScopes.includes('namespace') || !group.includes(TopologyGroupTypes.namespaces)) &&
-    (enabledScopes.includes('owner') || !group.includes(TopologyGroupTypes.owners))
-  );
-};
+export type TopologyGroupTypes = 'none' | Groups;
 
-export const getStepIntoNext = (current: FlowScope, allowedScopes: FlowScope[]): FlowScope | undefined => {
+export const getStepIntoNext = (current: FlowScope, scopes: ScopeConfigDef[]): FlowScope | undefined => {
   let next: FlowScope | undefined = undefined;
-  switch (current) {
-    case 'cluster':
-      next = 'zone';
-      break;
-    case 'zone':
-      next = 'host';
-      break;
-    case 'host':
-      next = 'resource';
-      break;
-    case 'namespace':
-      next = 'owner';
-      break;
-    case 'owner':
-      next = 'resource';
-      break;
-  }
-  if (!next || allowedScopes.includes(next)) {
-    return next;
+  const index = scopes.findIndex(sc => sc.id === current);
+  if (index >= 0 && index < scopes.length) {
+    next = scopes[index].id;
```

Comment:
step into doesn't work and I believe there's something missing here? (should it be `next = scopes[index + 1].id;` ?)
But that said, in [this PR](https://github.com/netobserv/network-observability-console-plugin/pull/550) I made the "scope graph" not linear on purpose (because the next logical step after node is pod, not namespace) .. so how to represent that with the new model? Should we add a new `next` field in the scopes definition?

Reply:
yeah I wanted to avoid that but I guess we don't have the choice if it's not linear there

Reply:
done in https://github.com/netobserv/network-observability-console-plugin/pull/634/commits/0c65c7f8690ce052f3c0cdf69d1c120780567cf5